integration-cli/docker_cli_pull_test.go

```diff
@@ -2,126 +2,104 @@ package main
 
 import (
 	"fmt"
-	"os/exec"
+	"regexp"
 	"strings"
 	"time"
 
-	"io/ioutil"
-
+	"github.com/docker/distribution/digest"
 	"github.com/go-check/check"
 )
 
-// See issue docker/docker#8141
-func (s *DockerRegistrySuite) TestPullImageWithAliases(c *check.C) {
-	repoName := fmt.Sprintf("%v/dockercli/busybox", privateRegistryURL)
+// TestPullFromCentralRegistry pulls an image from the central registry and verifies that the client
+// prints all expected output.
+func (s *DockerHubPullSuite) TestPullFromCentralRegistry(c *check.C) {
+	out := s.Cmd(c, "pull", "hello-world")
+	defer deleteImages("hello-world")
 
-	repos := []string{}
-	for _, tag := range []string{"recent", "fresh"} {
-		repos = append(repos, fmt.Sprintf("%v:%v", repoName, tag))
+	if !strings.Contains(out, "Using default tag: latest") {
+		c.Fatal("expected the 'latest' tag to be automatically assumed")
 	}
-
-	// Tag and push the same image multiple times.
-	for _, repo := range repos {
-		dockerCmd(c, "tag", "busybox", repo)
-		dockerCmd(c, "push", repo)
+	if !strings.Contains(out, "Pulling from library/hello-world") {
+		c.Fatal("expected the 'library/' prefix to be automatically assumed")
 	}
-
-	// Clear local images store.
-	args := append([]string{"rmi"}, repos...)
-	dockerCmd(c, args...)
-
-	// Pull a single tag and verify it doesn't bring down all aliases.
-	dockerCmd(c, "pull", repos[0])
-	dockerCmd(c, "inspect", repos[0])
-	for _, repo := range repos[1:] {
-		if _, _, err := dockerCmdWithError("inspect", repo); err == nil {
-			c.Fatalf("Image %v shouldn't have been pulled down", repo)
-		}
+	if expected := "Downloaded newer image for hello-world:latest"; !strings.Contains(out, expected) {
+		c.Fatalf("expected string %q in output\n%v", expected, out)
 	}
-}
-
-// pulling library/hello-world should show verified message
-func (s *DockerSuite) TestPullVerified(c *check.C) {
-	c.Skip("Skipping hub dependent test")
 
-	// Image must be pulled from central repository to get verified message
-	// unless keychain is manually updated to contain the daemon's sign key.
-
-	verifiedName := "hello-world"
-
-	// pull it
-	expected := "The image you are pulling has been verified"
-	if out, exitCode, err := dockerCmdWithError("pull", verifiedName); err != nil || !strings.Contains(out, expected) {
-		if err != nil || exitCode != 0 {
-			c.Skip(fmt.Sprintf("pulling the '%s' image from the registry has failed: %v", verifiedName, err))
-		}
-		c.Fatalf("pulling a verified image failed. expected: %s\ngot: %s, %v", expected, out, err)
+	digestExpr := regexp.MustCompile(`Digest: (.+)\n`)
+	if matches := digestExpr.FindAllStringSubmatch(out, -1); len(matches) != 1 {
+		c.Fatalf("expected exactly one image digest in the output, got %d", len(matches))
+	} else if _, err := digest.ParseDigest(matches[0][1]); err != nil {
+		c.Fatalf("invalid digest %q in output", matches[0][1])
```

Comment:
Probably just a nit (and because I think we don't use enough the power of `check`) — I'm writing this by memory so there might be a typo :sweat_smile: :

```go
matches = len(digestExpr.FindAllStringSubmatch(out, -1))
c.Check(matches, check.Equals, 1, check.Commentf("expected exactly one image digest in the output, got %d", matches))
_, err := digest.ParseDigest(matches[0][1])
c.Check(err, check.IsNil, check.Commentf("invalid digest %q in output", matches[0][1]))
```